src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let customJoin = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const index = this[i];
+
+      if (index !== null && index !== undefined) {
+        customJoin += index;
+      }
+
+      if (i !== this.length - 1) {
```

Comment:
I also suggest putting this.length in a variable, since we already use it twice, but I guess it's optional